streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java

```diff
@@ -43,16 +45,37 @@
 /**
  * {@code KStreamBuilder} provide the high-level Kafka Streams DSL to specify a Kafka Streams topology.
  *
- * @see TopologyBuilder
+ * @see org.apache.kafka.streams.processor.TopologyBuilder
  * @see KStream
  * @see KTable
  * @see GlobalKTable
+ * @deprecated Use {@link org.apache.kafka.streams.StreamsBuilder StreamsBuilder} instead
  */
-@InterfaceStability.Evolving
+@Deprecated
 public class KStreamBuilder extends TopologyBuilder {
 
     private final AtomicInteger index = new AtomicInteger(0);
 
+    private final InternalStreamsBuilder internalStreamsBuilder = new InternalStreamsBuilder();
+
+    private Topology.AutoOffsetReset translateAutoOffsetReset(final TopologyBuilder.AutoOffsetReset resetPolicy) {
+        if (resetPolicy == null) {
+            return null;
+        }
+        return resetPolicy == TopologyBuilder.AutoOffsetReset.EARLIEST ? Topology.AutoOffsetReset.EARLIEST : Topology.AutoOffsetReset.LATEST;
+    }
+
+    public KStreamBuilder() {
+        // TODO: we should refactor this to avoid usage of reflection
```

Comment:
As discussed, we can't keep as-is, because e.g. `KStreamImpl` was updated to accept an `InternalStreamBuilder` but not a `KStreamBuilder` anymore.